transport/rif_comms/exceptions.py

```diff
@@ -0,0 +1,116 @@
+from abc import ABC, abstractmethod
+from typing import Union
+
+from grpc import RpcError, StatusCode
+
+
+class ClientException(Exception, ABC):
+
+    @property
+    @abstractmethod
+    def get_code(self):
+        """
+        return the code property
+        """
+
+    @property
+    @abstractmethod
+    def get_message(self):
+        """
+        return the code property
+        """
+
+    def __str__(self):
+        message = super(ClientException, self).__str__()
+        return f"Code: {self.get_code}, Message: {message}"
+
+
+class InvalidArgumentException(ClientException):
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
+        super(InvalidArgumentException, self).__init__(message)
+
+    def get_code(self):
+        return self.code
+
+    def get_message(self):
+        return self.message
+
+
+class InternalException(ClientException):
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
+        super(InternalException, self).__init__(message)
+
+    def get_code(self):
+        return self.code
+
+    def get_message(self):
+        return self.message
+
+
+class NotFoundException(ClientException):
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
+        super(NotFoundException, self).__init__(message)
+
+    def get_code(self):
+        return self.code
+
+    def get_message(self):
+        return self.message
+
+
+class FailedPreconditionException(ClientException):
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
+        super(FailedPreconditionException, self).__init__(message)
+
+    def get_code(self):
+        return self.code
+
+    def get_message(self):
+        return self.message
+
+
+class TimeoutException(ClientException):
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
+        super(TimeoutException, self).__init__(message)
+
+    def get_code(self):
+        return self.code
+
+    def get_message(self):
+        return self.message
+
+
+EXCEPTION_MAPPING = {
+    StatusCode.DEADLINE_EXCEEDED: TimeoutException,
+    StatusCode.INTERNAL: InternalException,
+    StatusCode.INVALID_ARGUMENT: InvalidArgumentException,
+    StatusCode.NOT_FOUND: NotFoundException,
+    StatusCode.FAILED_PRECONDITION: FailedPreconditionException
+}
```

Comment:
Because is not needed in the client class, the mapper is used by the decorator now and that is in this file so we don't need the mapping on the other file.